File: src/python/_0011_Container_With_Most_Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans = 0
        width = len(height) - 1
        l, r = 0, width

        while l < r:
            if height[l] < height[r]:
                vol = height[l] * width
                l += 1
            else:
                vol = height[r] * width
                r -= 1

            width -= 1

            if vol > ans:
                ans = vol

        return ans

File: src/_0011_Container_With_Most_Water.c
int maxArea(int *height, int heightSize)
{
    int ans = 0, vol = 0, *rHeight = height + heightSize - 1;
    while (height < rHeight)
    {
        if (*height < *rHeight)
            vol = *(height++) * --heightSize;
        else
            vol = *(rHeight--) * --heightSize;
        if (vol > ans)
            ans = vol;
    }
    return ans;
}
